src/types/jsonrpc.ts

```diff
@@ -38,6 +38,22 @@ export const JSONRPCRequestStruct = object({
   params: optional(array())
 });
 
-export type JsonRPCRequest<T = unknown[]> = Infer<typeof JSONRPCRequestStruct> & {
-  params: T;
+export type JsonRPCRequest<T = unknown[]> = Omit<Infer<typeof JSONRPCRequestStruct>, 'params'> & {
+  params?: T;
 };
+
+export interface JsonRPCResponse<Result = unknown, Error = unknown> {
```

Comment:
Should we do a union type for this to separate when `result` and `error` are present?